platform/mds/mds/src/main/java/org/motechproject/mds/service/DefaultMotechDataService.java

```diff
@@ -317,8 +317,11 @@ public T revertFromTrash(Long trashId) {
             T newInstance = getClassType().newInstance();
 
             copyValuesFromRecord(newInstance, trashRecord);
+            newInstance = repository.create(newInstance);
```

Comment:
use create from motechdataservice, after your change we lost sending create event